src/scope.rs

```diff
@@ -29,13 +36,15 @@ impl Config {
 #[derive(Debug, Clone, Default)]
 pub struct Scope<'a> {
     /// Functions defined in this scope
-    defined: Set<&'a str>,
+    defined: Set<UniCase<&'a str>>,
     /// Defined by a file imported by `.`
-    directly_imported: Set<&'a str>,
+    directly_imported: Set<UniCase<&'a str>>,
     /// All the functions in scope
-    pub all: Set<&'a str>,
+    pub all: Set<UniCase<&'a str>>,
     /// All the files imported (directly and indirectly)
     files: Set<&'a Path>,
+    /// All defined or directly imported functions
+    directly_imported_or_defined: Set<UniCase<&'a str>>,
```

Comment:
I'm not a fan of duplicating knowledge, but meh, let it stay.

Reply:
Well, you said I should do it to avoid O(n^2) :)

Reply:
I said that *if* you need some preprocessing, just do it. In this particular example, this would also work

```rust
defined.contains(case_insensitive_name) || directly_imported.contains(case_insensitive_name)
```

Reply:
That's true. I was thinking of implementation with `get` instead of `contains`
```match self.directly_imported_or_defined.get(&case_insensitive_name)```
in this case, I can't just use `||`. Anyways it can be removed in this case.

Reply:
I remember playing around with or_else, but for some reason I didn't come up with this (obvious) snippet. Thanks